### PR TITLE
Fix vulnmanagement opname exceptions and inconsistencies for integration tests

### PR DESCRIPTION
**Pull request by pedrottimark**
## Description

TL;DR pattern for opname of GraphQL queries:

* plural: getNodeComponents for /main/vulnerability-management/node-components
* singular: getNodeComponent for:
    * side panel: /main/vulnerability-management/node-components?workflowState[0][t]=NODE_COMPONENT&workflowState[0][i]=id
    * single page: /main/vulnerability-management/node-component/id
* combined: getNodeComponentNODE for:
    * side panel: /main/vulnerability-management/node-components?workflowState[0][t]=NODE_COMPONENT&workflowState[0][i]=id&workflowState[1][t]=NODE
    * single page: /main/vulnerability-management/node-component/id/nodes

### Analysis of current behavior

| path segment     | plural entities    | singular entity   | prefix exception          | suffix          |
| :---             | :---               | :---              | :---                      | :---            |
| clusters         | getClusters        | getCluster        | getCluster_               | CLUSTER         |
| components       | getComponents      | getComponent      | getComponentSubEntity     | COMPONENT       |
| image-components | getImageComponents | getImageComponent | getComponentSubEntity     | IMAGE_COMPONENT |
| node-components  | getNodeComponents  | getComponent      | getNodeComponentSubEntity | NODE_COMPONENT  |
| cves             | getCves            | getCve            |                           | CVE             |
| cluster-cves     | getClusterCves     | getCve            |                           | CLUSTER_CVE     |
| image-cves       | getImageCves       | getCve            |                           | IMAGE_CVE       |
| node-cves        | getNodeCves        | getCve            |                           | NODE_CVE        |
| deployments      | getDeployments     | getDeployment     |                           | DEPLOYMENT      |
| images           | getImages          | getImage          |                           | IMAGE           |
| namespaces       | getNamespaces      | getNamespace      |                           | NAMESPACE       |
| nodes            | getNodes           | getNode           |                           | NODE            |
| policies         | getPolicies        | getPolicy         |                           | POLICY          |

### Primary exceptions in current behavior

To intercept and wait for relevant requests successfully, I tried most possibilities manually. On one side of the coin: what’s done is done. One the other side of the coin: simplify before next round of improvements to Vulnerability Management.

* The 4 **prefix exceptions** require an extra map for the test helper function:

    ```js
    function opnameForPrimaryAndSecondaryEntities(entitiesKey1, entitiesKey2) {
        return `${opnamePrefixForPrimaryAndSecondaryEntities[entitiesKey1]}${typeOfEntity[entitiesKey2]}`;
    }
    ```

    * `getCluster_NAMESPACE` given arguments: `clusters` and `namespaces` although underscore is nice for readability
    * `getComponentSubEntityIMAGE` given arguments: `components` and `images`
    * `getComponentSubEntityIMAGE` given arguments: `image-components` and `images`
    * `getComponentSubEntityNODE` given arguments: `node-components` and `nodes`

* Without exceptions, here is a simpler test helper function:

    ```js
    function opnameForPrimaryAndSecondaryEntities(entitiesKey1, entitiesKey2) {
        return `${opnameForEntity[entitiesKey1]}${typeOfEntity[entitiesKey2]}`;
    }
    ```

    * `getClusterNAMESPACE`
    * `getComponentIMAGE` given arguments: `components` and `images`
    * `getImageComponentIMAGE` given arguments: `image-components` and `images`
    * `getComponentNODE` given arguments: `node-components` and `nodes` see next section

### Secondary inconsistencies in current behavior

While I tried most possibilities manually, it looked at first like I had made a mistake to see getCve after getClusterCves, and so on. After I repeated the steps, it looked like a bug.

* Partly consistent: forward-looking image-components and node-components have strings which differ from each other and also from backward-looking components, except for node-components **singular** getComponent.
* Inconsistent: forward-looking cluster-cves, image-cves, node-cves have **plural** and **suffix** strings which do differ differ from each other and also from backward-looking cves, but not for **singular** strings. It is misleading for singular request opname and main query name to look as cve type has not been specified for postgres data store.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Edited integration tests

### manual testing

TODO Compare requests and results on infra demo to staging demo

### integration testing

1. Fix 4 primary exceptions
    in component code, but not test helpers, to verify test coverage
    * clusters: 3 tests fail for namespaces, deployments, node
    * non-postgress components: 1 test fails for images, deployments?
    * postgress image-components: 2 tests fail for images, deployments
    * postgress node-components: 1 test fails for nodes
    in test helpers

2. Fix 4 secondary inconsistencies
    * in component code, but not test helpers, to verify test coverage
    * in test helpers

### CODE REVIEW

**Review by vjwilson — APPROVED** [review]
/lgtm
